data-avro/src/main/java/com/linkedin/data/avro/DefaultDataToAvroConvertCallback.java

```diff
@@ -215,6 +215,17 @@ protected Object translateField(List<Object> path, Object fieldValue, RecordData
           (_options.getOptionalDefaultMode() == OptionalDefaultMode.TRANSLATE_TO_NULL && isTranslatedUnionMember) ||
           (fieldValue == Data.NULL));
     }
+    else if (fieldValue == null)
+    {
+      // If the default specified at parent level doesn't specify a value for the field, use the default specified at
+      // field level.
+      fieldValue = field.getDefault();
+      if (fieldValue == null)
```

Comment:
if this field is a record,  and this field doesn't have the defaults. So `fieldValue` will be null.

But what is each field of this record it self have defaults?  Does that mean this record field implictly have a default?

Are we allowing that?

Reply:
If this field is a record without default. Then the parent should've specified a default value for this field (even if all its nested fields have default). So there is no concept of implicit default. 
Think of it this way: The default value specified at any level should be a valid representation of the schema. So there should be an entry for all required fields at each level (either from parent or at the field level). No assumptions can be made about the nested levels.